src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFsDriver.Linux.cs

```diff
@@ -302,15 +315,25 @@ protected internal override WaitForEventResult WaitForEvent(int pinNumber, PinEv
         private void SetPinEventsToDetect(int pinNumber, PinEventTypes eventTypes)
         {
             string edgePath = Path.Combine(GpioBasePath, $"gpio{pinNumber + s_pinOffset}", "edge");
-            string stringValue = PinEventTypeToStringValue(eventTypes);
-            File.WriteAllText(edgePath, stringValue);
+            if (File.Exists(edgePath)) // Only exists for input pins!
```

Comment:
Does it even make sense to allow detecting events for output pins? If not, then we should probably add this validation directly on GpioController so that no driver will allow this check.

Reply:
I don't know if there are gpio drivers around which will support this functionality. But at least for the Beaglebone linux distributions with SysFs I haven't seen it jet.